include/fmt/core.h

```diff
@@ -1131,6 +1132,7 @@ template <typename Context> class basic_format_arg {
 
   friend class basic_format_args<Context>;
   friend class internal::arg_map<Context>;
+  friend class dynamic_format_arg_store<Context>;
```

Comment:
Is this necessary?

Reply:
Yes, `basic_format_arg` is constructed from `dynamic_format_arg_store::emplace_arg()` and needs to access private ctor. Also there's a need to update `named_args` with new size and pointer to array of `named_arg_info` after potential relocation.

Reply:
done